python_modules/libraries/dagster-dbt/dagster_dbt/cli/resources_v2.py

```diff
@@ -827,7 +832,9 @@ def stream_raw_events(self) -> Iterator[DbtCliEventMessage]:
                 sys.stdout.flush()
 
         # Ensure that the dbt CLI process has completed.
-        self.process.wait()
+        self.is_successful()
+        if not self.is_successful() and self.raise_on_error:
+            raise Failure
```

Comment:
Based on [this thread](https://elementl-workspace.slack.com/archives/C03A0D72A6T/p1688142008101359) with @alangenfeld , it seems like the main reason to use `Failure` is if we want to add metadata and control the retries, but those don't seem to be the case here. It would be better to have a more specific error type.  Then users could catch it if they want to do their own special handling.

The existing error-handling code appears to distinguish between errors that cause the dbt CLI to fail unexpectedly, and errors that the dbt CLI handles but also reports. I don't know the background enough to be able to say whether it's important that we distinguish between these, so can probably leave that out for now, but just mentioning here for posterity.

Reply:
I will just use `DagsterDbtCliRuntimeError` since that already exists.